httpcore5/src/main/java/org/apache/hc/core5/http/io/entity/EntityUtils.java

```diff
@@ -97,6 +97,11 @@ public static void consume(final HttpEntity entity) throws IOException {
         }
     }
 
+    private static int getCheckedContentLength(final HttpEntity entity) {
+        final int contentLength = (int) Args.checkContentLength(entity);
```

Comment:
@garydgregory I am not a very big fan of shared code at all cost. I would very much prefer to keep `Args.checkContentLength` and similar argument checks at the very beginning of each method, and not hidden somewhere deep inside a static method. Otherwise looks good to me.

Reply:
I agree, they should be on public methods.

Reply:
@ok2c I moved the checks to the start of methods. 

@michael-o I did not add public methods to keep the API surface minimal.

Reply:
@garydgregory What I actually mean was that `Args#checkContentLength` should be at the beginning of the method. `getCheckedContentLength` method could be wherever it needs to be. Please decouple `Args#checkContentLength`  and `getCheckedContentLength`. They are unrelated.

Reply:
@ok2c Sorry for the misunderstanding, I just committed changes for this request.